### ISSUE TITLE
Throw error instead of exiting silently

### ISSUE THREAD

**Issue by DanielHabenicht** [enowars/EnoEngine#142]
https://github.com/enowars/EnoEngine/blob/950a3da5a8df6465c3c732f81a7269dbe002b026/EnoEngine/Program.cs#L34-L38

is just one of example of silently exiting.

It should throw errors instead: 

```csharp
throw new Exception("Config (ctf.json) does not exist.");
```

**Comment by Trolldemorted** [comment]
Well it is not silent, it prints a very clear message why it exits early. Last year students complained when they saw stacktraces, so we switched to messages only :P

Should we switch to stderr and return a non-zero exit code?

**Comment by Savallator** [comment]
Yeah, Stacktraces do not really help there... also keep in mind that the engine is not inside docker. The Console.Writelines plop directly into your shell, it is really hard to miss.

**Comment by DanielHabenicht** [comment]
What? Why did they complain about stack traces? That's normal in IT. 😂
I mean if they don't learn it here, where else? 

![image](https://user-images.githubusercontent.com/13590797/115992284-25716800-a5cd-11eb-87e3-a69328625875.png)

It is displaying the same message just with more information and in Visual Studio it even opens a popup that something went wrong instead of just exiting silently. 

![image](https://user-images.githubusercontent.com/13590797/115992402-dd067a00-a5cd-11eb-85ba-8266dddb9900.png)

I get it that for newcomers it may make things more complicated, but programming is based on standards and these should be adhered to.

**Comment by Trolldemorted** [comment]
>What? Why did they complain about stack traces? That's normal in IT. 😂

To be fair, that's normal for crappy programs, but not the general state of IT. Imagine linux utils like `cat` printing stack traces if they can't open a file, that would be a horrible UX. Instead, they print a concise errormessage `cat: ./foo: No such file or directory` and exit.

**Comment by ldruschk** [comment]
But these tools do exit with a non-zero exit code

**Comment by Trolldemorted** [comment]
>But these tools do exit with a non-zero exit code

Changing that should be easy, just `return 1` I think

**Comment by DanielHabenicht** [comment]
We could also wrap the program into an ExceptionHandler as you already do it in the lines below. 

This way throughout the program you could still use Exceptions and then write a nice output with the Release build. 
Something along: 

```csharp
catch (Exception e)
{
#if (DEBUG)
    throw e;
#else
    Console.WriteLine(e.Message);
    return 1;
#endif
}
finally
{
    mutex?.Close();
}
return 0;
```

**Comment by Savallator** [comment]
But you will never actually need a Stacktrace in this particular case. It is just stupid to make it return a useless stacktrace instead of a single message to exactly tell you what the problem is.

**Comment by DanielHabenicht** [comment]
I agree that you don't need an actual Stacktrace in the case I provided. 

But just to name an example where it actually hurts:
https://github.com/enowars/EnoEngine/blob/950a3da5a8df6465c3c732f81a7269dbe002b026/EnoEngine/Program.cs#L66-L70 
Here the child Exception is swallowed and will never be displayed. If the service for a checker is not reachable the only message displayed will be `Configuration is invalid.`, whereas with proper Exception: 
![image](https://user-images.githubusercontent.com/13590797/115999480-409fa000-a5ec-11eb-8cc2-c3f0bdc2e5ef.png)

**Comment by Savallator** [comment]
I don't really get what you are trying to do. The code you referenced belongs to the checking of the config file. The stacktrace there does not contain any useful information, and the exception message is already printed. The HTTPRequestException you posted is something else entirely. The case when a checker is not reachable should also not throw an Exception, but should exit with the appropriate Error or continue for testing, as discussed in #144. What would an exception with a stacktrace into the HTTPClient actually help anyone, when it is as simple as a connection being refused?

**Comment by DanielHabenicht** [comment]
This issue is about the general practice of not throwing errors but just silently exiting (during development). 
As I said there no useful message is provided because it just prints to console, furthermore I am not even getting the whole error because it just prints `Configuration is invalid.` although the issue is with the service not reachable. A Stacktrace would have helped resolve this issue faster.

I tried to run the Engine but kept on running into issues. 
This is something that would have helped me fix the mistakes I did during startup much faster. (e.g. not running a checker before starting the engine)

Are you in Mumble? Maybe it is easier to just speak about it.
